src/app/components/Search/index.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { useState, useEffect } from 'react';
+import { Input } from '@cfxjs/react-ui';
+import { SearchIcon } from '../SearchIcon/Loadable';
+import { DeleteIcon } from '../DeleteIcon/Loadable';
+type SearchProps = {
+  children?: React.ReactNode;
+  outerClassname?: string;
+  inputClassname?: string;
+  iconColor?: string;
+  placeholderText?: string;
+  onEnterPress?: (
+    e: React.KeyboardEvent<HTMLDivElement>,
+    value?: string,
+  ) => void;
+  onChange?: (value: string) => void;
+  val?: string;
+};
+type NativeAttrs = Omit<React.HTMLAttributes<any>, keyof SearchProps>;
+export declare type Props = SearchProps & NativeAttrs;
+
+export const Search = ({
+  outerClassname,
+  placeholderText,
+  inputClassname,
+  onEnterPress,
+  iconColor,
+  onChange,
+  val,
+  ...props
+}: Props) => {
+  const [value, setValue] = useState(val);
+  const [type, setType] = useState('search');
+  const [clickable, setClickable] = useState(false);
+  const onIconRightClickHander = () => {
+    if (type === 'delete') {
+      setValue('');
+    }
+  };
+  const onChangeHandler = e => {
+    setValue(e.target.value);
+    onChange && onChange(e.target.value);
+  };
+  useEffect(() => {
+    if (value) {
+      setType('delete');
+      setClickable(true);
+    } else {
+      setType('search');
+      setClickable(false);
+    }
+  }, [value]);
```

Comment:
may not use useEffect for type and clickable change, use like:
```
const type = value ? 'delete' : 'search';
const clickable = !!value;
```

Reply:
done